Source/WebCore/page/Navigation.cpp

```diff
@@ -795,4 +794,12 @@ bool Navigation::dispatchDownloadNavigateEvent(const URL& url, const String& dow
     return innerDispatchNavigateEvent(NavigationNavigationType::Push, WTFMove(destination), downloadFilename);
 }
 
+// https://html.spec.whatwg.org/multipage/nav-history-apis.html#inform-the-navigation-api-about-aborting-navigation
+void Navigation::informAboutCanceledNavigation()
```

Comment:
This is sort of in the “uncanny valley” of function names, similar to, but not the same as, the name in the specification, and without an object, the phrase “inform about” isn’t so great. I think we can do a little better with the name. One way is to just write it all out, `informNavigationAPIAboutAbortingNavigation`. Or we could just leave it named `abortOngoingNavigation` and remove the argument from that function? Or overload `abortOngoingNavigation` to have this be a version without an argument? Or name it `abortOngoingNavigateEvent`? Maybe there are some even better names.

Reply:
Note that in the spec the algorithm is a bit more complicated than in this implementation, since it has some logic for  the window event loop, which does not seem to be needed so far. Having said that, indeed the method name is quite generic and maybe a bit verbose. I'd rather leave the abortOngoingNavigation as-is, since it is not public. In the end I came up with abortOngoingNavigationIfNeeded, which I think is pretty clear, hope you like it.